foreshadow/utils/default_estimator_factory.py

```diff
@@ -0,0 +1,121 @@
+"""Estimator factory and supported estimators."""
+
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+from sklearn.linear_model import LinearRegression, LogisticRegression
+from sklearn.neural_network import MLPClassifier, MLPRegressor
+from sklearn.svm import LinearSVC, LinearSVR
+
+
+class EstimatorFamilyMixin:
+    """Mixin class that allows an instance to retrieve an estimator.
+
+    Retrieve is based on the type of the problem (classification or
+    regression).
+    """
+
+    def get_estimator_by_type(self, problem_type):
+        """Retrieve an estimator based on the problem type.
+
+        Args:
+            problem_type: either classification or regression
+
+        Returns:
+            an estimator of a particular model family
+
+        """
+        return self.estimators[problem_type]
+
+
+class LinearEstimatorFamily(EstimatorFamilyMixin):
+    """Linear model estimator family.
+
+    Estimators include LinearRegression and LogisticRegression.
+    """
+
+    def __init__(self):
+        self.estimators = {
+            "classification": LogisticRegression(),
+            "regression": LinearRegression(),
+        }
+
+
+class SVMEstimatorFamily(EstimatorFamilyMixin):
+    """SVM model estimator family.
+
+    Estimators include LinearSVC and LinearSVR.
+    """
+
+    def __init__(self):
+        self.estimators = {
+            "classification": LinearSVC(),
+            "regression": LinearSVR(),
```

Comment:
This way of implementation creates an instance of all estimators even only one of them will be used during a specific Foreshadow run. It might be better to only register estimator class and create an instance of it in the get_estimator method of the factory.